packages/grid/_modules_/grid/models/api/gridEditingApi.ts

```diff
@@ -20,19 +24,28 @@ export interface GridEditRowApi {
    */
   getEditRowsModel: () => GridEditRowsModel;
   /**
-   * Sets the mode of a cell.
-   * @param {GridRowId} id The id of the row.
-   * @param {string} field The field to change the mode.
-   * @param {GridCellMode} mode Can be: `"edit"`, `"view"`.
+   * Controls if a cell is editable.
+   * @param {GridCellParams} params The cell params.
+   * @returns {boolean} A boolean value determining if the cell is editable.
    */
-  setCellMode: (id: GridRowId, field: string, mode: GridCellMode) => void;
+  isCellEditable: (params: GridCellParams) => boolean;
   /**
-   * Gets the mode of a cell.
-   * @param {GridRowId} id The id of the row.
-   * @param {string} field The field to get the mode.
-   * @returns {GridCellMode} Returns `"edit"` or `"view"`.
+   * Sets the value of the edit cell.
+   * Commonly used inside the edit cell component.
+   * @param {GridEditCellValueParams} params Contains the id, field and value to set.
+   * @param {React.SyntheticEvent} event The event to pass forward.
    */
-  getCellMode: (id: GridRowId, field: string) => GridCellMode;
+  setEditCellValue: (params: GridEditCellValueParams, event?: MuiBaseEvent) => void;
+  /**
+   * @ignore - do not document.
+   */
+  unstable_setEditCellProps: (params: GridEditCellPropsParams) => void;
+}
+
+/**
+ * The row editing API interface.
+ */
+export interface GridRowEditingApi extends GridEditingSharedApi {
```

Comment:
It extends `GridEditingSharedApi` because I want to create a separate page for row editing and another for cell editing. That way, I can add the API widget with all methods that make sense for row editing. The same for cell editing.